### DISCUSSION TITLE
Text replacement

### DISCUSSION THREAD

**Discussion by mathanimas** [notaapp/talk#89] [Feature Requests]
Text snippets specified in the Mac system preferences (Keyboard → Text → Replace/With) are not working. It would make Nota much more useful if these shortcuts were respected there, too.

**Comment by erusev** [comment]
Hi @mathanimas and thanks for the request! This is sth I've been researching (and wanting to use) for a while, and I have a couple of questions about the feature, perhaps you can help me.

1. Is there an easy way to sync replacement entries — bring them with you when you move to a new device or reinstall macOS?
2. Is there an easy way to export replacement entries? I'm concerned with the scenarios where I've spent hours adding entries (over some time), and after a while, I find a new app that does text replacements better, but it's too hard to switch because I can't export.

**Comment by led** [comment]
> Is there an easy way to export replacement entries? I'm concerned with the scenarios where I've spent hours adding entries (over some time), and after a while, I find a new app that does text replacements better, but it's too hard to switch because I can't export.

A starting point would be…

```
#!/bin/sh

# Export system keyboard expansions
/usr/libexec/PlistBuddy -c "Print :NSUserDictionaryReplacementItems" ~/Library/Preferences/.GlobalPreferences.plist | sed -E '/replace/!d;N;s/ {8}//g;s/(replace|with) = //g;s/\n/","/g;s/(.*)/["\1"]/g'

```

…and then parse that to whatever format you fancy.

**Comment by mathanimas** [comment]
Hi @erusev, thanks for the quick reply. I’ll gladly try to help.

> 1. Is there an easy way to sync replacement entries — bring them with you when you move to a new device or reinstall macOS?

This happens automatically. I believe I have never had to think about that. I suppose that Apple uses iCloud to sync that.
However, I can confirm that they are also contained in ~/Library/Preferences/.GlobalPreferences.plist.

Maybe this helps.

> 2. Is there an easy way to export replacement entries? I'm concerned with the scenarios where I've spent hours adding entries (over some time), and after a while, I find a new app that does text replacements better, but it's too hard to switch because I can't export.

I can confirm that what @led just posted works at my Mac.

**Comment by erusev** [comment]
Great info! Makes me more excited about the feature 😊 I'll see what we can do to add support for it in Nota.

**Comment by Manu1400** [comment]
@mathanimas @erusev Install https://github.com/yannbertrand/macos-defaults, then run `defaults read NSGlobalDomain NSUserDictionaryReplacementItems` to get all data.
